Screenshot Maker.cpp

```diff
@@ -23,18 +23,29 @@ int APIENTRY wWinMain(HINSTANCE hInstance,
 					  LPWSTR    lpCmdLine,
 					  int       nCmdShow)
 {
+	bool allscreens = false;
+	for (int i = 1; i < __argc; i++)
+	{
+		if (wcscmp(__wargv[i], L"-all") == 0 || wcscmp(__wargv[i], L"-a") == 0)
+		{
+			allscreens = true;
+		}
+	}
+
 	GdiplusStartupInput gdiplusStartupInput;
 	ULONG_PTR gdiplusToken;
 	GdiplusStartup(&gdiplusToken, &gdiplusStartupInput, NULL);
 
 	HWND desktop = GetDesktopWindow();
-	HDC desktopdc = GetDC(desktop);
+	HDC desktopdc = allscreens ? CreateDC(TEXT("DISPLAY"), NULL, NULL, NULL): GetDC(desktop);
 	HDC mydc = CreateCompatibleDC(desktopdc);
-	int width = GetSystemMetrics(SM_CXSCREEN);
-	int height = GetSystemMetrics(SM_CYSCREEN);
+	int width = GetSystemMetrics(allscreens? SM_CXVIRTUALSCREEN :SM_CXSCREEN);
+	int height = GetSystemMetrics(allscreens ? SM_CYVIRTUALSCREEN : SM_CYSCREEN);
+	int top = allscreens?GetSystemMetrics(SM_YVIRTUALSCREEN):0;
+	int left = allscreens ? GetSystemMetrics(SM_XVIRTUALSCREEN):0;
 	HBITMAP mybmp = CreateCompatibleBitmap(desktopdc, width, height);
 	HBITMAP oldbmp = (HBITMAP)SelectObject(mydc, mybmp);
-	BitBlt(mydc,0,0,width,height,desktopdc,0,0, SRCCOPY|CAPTUREBLT);
+	BitBlt(mydc,0,0,width,height,desktopdc,left,top, SRCCOPY|CAPTUREBLT);
 	SelectObject(mydc, oldbmp);
 
 	bool defaultfn = true;
```

Comment:
How about avoiding an additional loop over argv by moving this block (lines 35-56) after command line parsing?

Reply:
Okay, I can do that. Those loops are slightly different (different end index). I'll implement a check for missing arguments then (missing filename, missing quality etc.) and output error messages.

I will need AttachConsole() to output error messages, since the project is targeting the Windows Subsystem. 
https://learn.microsoft.com/en-us/windows/console/attachconsole

Reply:
> I will need AttachConsole() to output error messages, since the project is targeting the Windows Subsystem.

I don't mind if you leave it as is. My idea was that if somebody wants the console output, they can compile it targeting the console subsystem, but I personally needed to have no console window for my use case. Perhaps a better default would be to target the console subsystem, but in any case, it's out of scope for your PR of adding a new switch.